### PR TITLE
[Commands]: Fix block editor commands availability

### PR DESCRIPTION
**Pull request by ntsekouras**
## What?
Part of: https://github.com/WordPress/gutenberg/issues/53921

> By https://github.com/WordPress/gutenberg/pull/52509, commands related to blocks were added. However, you will see commands that are not expected to be used with the selected block.

This PR handles:

- Move to: Should not be displayed when there is only one block or when the move is locked
- Add after / Add before: When a default block cannot be inserted, it should not be displayed. For example, this is the case when the paragraph block is excluded by the `allowed_block_types_all hook`.
- Remove: Should not be displayed when `canRemoveBlocks` is false.
- Duplicate: Should not be displayed when the block [cannot be duplicated](https://github.com/WordPress/gutenberg/blob/a9252ece268e0a458016d65d302440c46db9b044/packages/block-editor/src/components/list-view/block-select-button.js#L165-L173).

## Testing Instructions
1. Test the above commands that are available only when they meet the respective conditions.

### CODE REVIEW

**Review by draganescu — APPROVED** [review]
Tested and works great, thank you @ntsekouras 🙏🏻

File: packages/block-editor/src/components/use-block-commands/index.js [hunk 1]
```diff
@@ -229,6 +176,21 @@ const useActionsCommands = () => {
...
+	);
+	const canDuplicate = blocks.every( ( block ) => {
+		return (
+			!! block &&
+			hasBlockSupport( block.name, 'multiple', true ) &&
+			canInsertBlockType( block.name, rootClientId )
+		);
+	} );
+	const canRemove = canRemoveBlocks( clientIds, rootClientId );
+	const canMove = canMoveBlocks( clientIds, rootClientId );
```
Comment (t-hamano) [inline packages/block-editor/src/components/use-block-commands/index.js:192]: I think we also need to check that there is more than one block, just as [the Move to action on the block toolbar does](https://github.com/WordPress/gutenberg/blob/3a7404dd80067715da65caffb7f830b3ca218617/packages/block-editor/src/components/block-settings-menu-controls/index.js#L93).

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR! It appears to be mostly fine, but I would like to comment on a few points of concern.

File: packages/block-editor/src/components/use-block-commands/index.js [hunk 2]
```diff
@@ -242,49 +204,73 @@ const useActionsCommands = () => {
...
+					const clientId = Array.isArray( clientIds )
+						? clientIds[ 0 ]
+						: clientId;
+					insertBeforeBlock( clientId );
+				},
+				icon: add,
+			}
+		);
+	}
+	if ( canRemove ) {
```
Comment (t-hamano) [inline packages/block-editor/src/components/use-block-commands/index.js:234]: I think we also need to check that there is more than one block, just as [the Move to action on the block toolbar does](https://github.com/WordPress/gutenberg/blob/3a7404dd80067715da65caffb7f830b3ca218617/packages/block-editor/src/components/block-settings-menu-controls/index.js#L93).

File: packages/block-editor/src/components/use-block-commands/index.js [hunk 3]
```diff
@@ -242,49 +204,73 @@ const useActionsCommands = () => {
...
 			icon: copy,
-		},
-	].map( ( { name, label, callback, icon } ) => {
-		return {
-			name: 'core/block-editor/action-' + name,
-			label,
-			icon,
+		} );
+	}
+	if ( canMove ) {
```
Comment (t-hamano) [inline packages/block-editor/src/components/use-block-commands/index.js:250]: I think we also need to check that there is more than one block, just as [the Move to action on the block toolbar does](https://github.com/WordPress/gutenberg/blob/3a7404dd80067715da65caffb7f830b3ca218617/packages/block-editor/src/components/block-settings-menu-controls/index.js#L93).

Comment (ntsekouras) [reply packages/block-editor/src/components/use-block-commands/index.js:250]: I'll land this one to rebase the other related PRs and will open a follow up for this. Thanks for testing and spotting this Aki!

Comment (ntsekouras) [reply packages/block-editor/src/components/use-block-commands/index.js:250]: Follow up here: https://github.com/WordPress/gutenberg/pull/54049

**Pull request merged**